### PR TITLE
make `FunctionModule`'s `avg`, `max`, `min` & `sum` functions nullable (consumer opt-in).

### PR DESCRIPTION
**Pull request by igalklebanov**
These aggregate functions are nullable, in some edge cases.

Current types:

a. are a lie, if some edge cases are possible in consumer's queries / state of database.
b. do not allow our consumers to state `null` as a possible return type.

Let's help our consumers write robust code, by allowing them to opt-in for the stricter nullable form.

Should fit smooth like 🧈 with recently merged `.coalesce(...)` function.

----

- [x] implement type changes.
- [x] fix typings tests.

### CODE REVIEW

File: src/query-builder/function-module.ts [hunk 1]
```diff
@@ -142,14 +145,19 @@ export class FunctionModule<DB, TB extends keyof DB> {
...
+    >
   >(
     column: C
   ): AggregateFunctionBuilder<
     DB,
     TB,
-    ExtractTypeFromReferenceExpression<DB, TB, C, O>
+    ExtractTypeFromReferenceExpression<DB, TB, C, O> | null extends O
+      ? null
+      : never
```
Comment (igalklebanov) [inline src/query-builder/function-module.ts:160]: Current types allow consumers to state `O` only when column is a `DynamicReference`. Let's open this up, so the consumer can tell us whether `O` is nullable, even when column is a `StringReference`, but let's still restrict return type to the type provided for that column in `Database`.

File: src/query-builder/function-module.ts [hunk 2]
```diff
@@ -211,9 +225,12 @@ export class FunctionModule<DB, TB extends keyof DB> {
    *   .select(sum<number>('price').as('total_price'))
    *   .execute()
    * ```
+   *
+   * It is highly recommended to include null in the output type union and handle
+   * null values in post-execute code, or wrap the function with a coalesce function.
    */
   sum<
-    O extends number | string | bigint,
+    O extends number | string | bigint | null = number | string | bigint,
```
Comment (igalklebanov) [inline src/query-builder/function-module.ts:233]: User can opt-in for nullable `O`.

File: src/query-builder/function-module.ts [hunk 3]
```diff
@@ -60,14 +60,15 @@ export class FunctionModule<DB, TB extends keyof DB> {
    * Calls the `avg` function for the column given as the argument.
    *
    * If this is used in a `select` statement the type of the selected expression
-   * will be `number | string` by default. This is because Kysely can't know the
-   * type the db driver outputs. Sometimes the output can be larger than the
-   * largest javascript number and a string is returned instead. Most drivers
-   * allow you to configure the output type of large numbers and Kysely can't
-   * know if you've done so.
+   * will be `number | string | null` by default. This is because Kysely can't know
```
Comment (koskimas) [inline src/query-builder/function-module.ts:63]: This isn't true is it? The default is now `number | string` with a user opt-in of including the `null`?

File: src/query-builder/function-module.ts [hunk 4]
```diff
@@ -143,11 +155,11 @@ export class FunctionModule<DB, TB extends keyof DB> {
    * Calls the `count` function for the column given as the argument.
    *
    * If this is used in a `select` statement the type of the selected expression
-   * will be `number | string | bigint` by default. This is because Kysely can't
-   * know the type the db driver outputs. Sometimes the output can be larger than
-   * the largest javascript number and a string is returned instead. Most drivers
-   * allow you to configure the output type of large numbers and Kysely can't
-   * know if you've done so.
+   * will be `number | string | bigint | null` by default. This is because Kysely
```
Comment (koskimas) [inline src/query-builder/function-module.ts:158]: Same here? Default is actually `number | string | bigint`.

File: src/query-builder/function-module.ts [hunk 5]
```diff
@@ -250,11 +285,12 @@ export class FunctionModule<DB, TB extends keyof DB> {
    * Calls the `sum` function for the column given as the argument.
    *
    * If this is used in a `select` statement the type of the selected expression
-   * will be `number | string | bigint` by default. This is because Kysely can't
-   * know the type the db driver outputs. Sometimes the output can be larger than
-   * the largest javascript number and a string is returned instead. Most drivers
-   * allow you to configure the output type of large numbers and Kysely can't
-   * know if you've done so.
+   * will be `number | string | bigint | null` by default. This is because Kysely
```
Comment (koskimas) [inline src/query-builder/function-module.ts:288]: Here too, null is not in the default.

File: src/query-builder/function-module.ts (hunk 3 above)
Comment (igalklebanov) [reply src/query-builder/function-module.ts]: Good catch! forgot about aligning the docs.. 😢

File: src/query-builder/function-module.ts (hunk 4 above)
Comment (igalklebanov) [reply src/query-builder/function-module.ts]: Good catch! forgot about aligning the docs.. 😢

File: src/query-builder/function-module.ts (hunk 5 above)
Comment (igalklebanov) [reply src/query-builder/function-module.ts]: Good catch! forgot about aligning the docs.. 😢

**Comment by igalklebanov** [comment]
Sorry about docs, hope its fine now.. 😅

**Comment by koskimas** [comment]
Thanks!

**Pull request merged**